### DISCUSSION TITLE
Question: Parsing Source File as `FileDescriptorProto`

### DISCUSSION THREAD

**Discussion by XAMPPRocky** [tokio-rs/prost#486] [Q&A]
Hey, I want to do more advanced codegen than `prost-build` currently provides with `ServiceGenerator`. I want to be able to generate a second configuration that is more Rust friendly, and I want to be able to set attributes based on what the names, types, and values of the different messages and fields are.

`FileDescriptorProto` has this information, however it doesn't like there is a public way to get `FileDescriptorProto` from a source `.proto` file. Would it be possible to have a public way to parse Protobuf source files?

**Comment by andrewhickman** [comment]
I have developed a crate for this: [protox](https://crates.io/crates/protox)
